tests/e2e/objects/views/ArtifactsView.ts

```diff
@@ -0,0 +1,246 @@
+
+    const clusterItem = clusterLabel
+      ? kafkaClusterQuickpick.items.filter({ hasText: clusterLabel }).first()
+      : kafkaClusterQuickpick.items.first();
+    await clusterItem.click();
+    await this.clickSwitchToFlinkArtifacts();
+  }
+
+  /**
+   * Upload a Flink artifact JAR file to Confluent Cloud.
+   * Clicks the upload button in the view title area, navigates through the quickpick steps,
+   * and selects the specified JAR file.
+   * @param electronApp - The Electron application instance
+   * @param filePath - The path to the JAR file to upload
+   * @returns The name of the uploaded artifact
+   */
+  async uploadFlinkArtifact(electronApp: ElectronApplication, filePath: string): Promise<string> {
+    await this.initiateUpload();
+    await this.selectJarFile(electronApp, filePath);
+    const artifactName = await this.enterArtifactName(filePath);
+    await this.confirmUpload();
+    await this.waitForUploadSuccess();
+    return artifactName;
+  }
+
+  /**
+   * Click the upload button to initiate the artifact upload flow.
+   */
+  private async initiateUpload(): Promise<void> {
+    const uploadButton = this.page.locator(
+      'a.action-label.codicon.codicon-cloud-upload[aria-label="Upload Flink Artifact to Confluent Cloud"]',
+    );
+    await uploadButton.click();
+
+    const quickpick = new Quickpick(this.page);
+    await expect(quickpick.locator).toBeVisible();
+    await expect(quickpick.items).not.toHaveCount(0);
+  }
+
+  /**
+   * Select the JAR file to upload using the system file dialog.
+   * @param electronApp - The Electron application instance
+   * @param filePath - The path to the JAR file
+   */
+  private async selectJarFile(electronApp: ElectronApplication, filePath: string): Promise<void> {
+    const quickpick = new Quickpick(this.page);
+    const selectedJarFileItem = quickpick.items.filter({ hasText: "3. Select JAR File" }).first();
+    await expect(selectedJarFileItem).toBeVisible();
+
+    await stubDialog(electronApp, "showOpenDialog", {
+      filePaths: [filePath],
+    });
+
+    await selectedJarFileItem.click();
+  }
+
+  /**
+   * Enter the artifact name in the quickpick input.
+   * @param filePath - The path to the JAR file (used to generate the name)
+   * @returns The full artifact name with random suffix
+   */
+  private async enterArtifactName(filePath: string): Promise<string> {
+    const quickpick = new Quickpick(this.page);
+    const artifactItem = quickpick.items.filter({ hasText: "4. Artifact Name" }).first();
+    await expect(artifactItem).toBeVisible();
+    await artifactItem.click();
+
+    // Although this resource may be cleaned up, we append a random string to avoid name conflicts during development
+    const randomSuffix = Math.random().toString(36).substring(2, 8);
+    const baseFileName = path.basename(filePath, ".jar");
+    const fullArtifactName = `${baseFileName}-${randomSuffix}`;
+
+    await this.page.keyboard.type(fullArtifactName);
```

Comment:
Keyboard events... but in this case, I think it would be the most common pathway